backend/mach.ml

```diff
@@ -179,12 +179,22 @@ let rec instr_iter f i =
             | Ibeginregion | Iendregion) ->
         instr_iter f i.next
 
+let operation_is_pure = function
+  | Icall_ind | Icall_imm _ | Itailcall_ind | Itailcall_imm _
+  | Iextcall _ | Istackoffset _ | Istore _ | Ialloc _
+  | Iintop(Icheckbound) | Iintop_imm(Icheckbound, _) | Iopaque -> false
+  | Ibeginregion | Iendregion -> false
+  | Iprobe _ | Iprobe_is_enabled _-> false
```

Comment:
Isn't `Iprobe_is_enabled` pure?

Reply:
(If so, `ocaml/asmcomp/mach.ml` needs
to be updated as well.)

Reply:
`Iprobe_is_enabled` is like Iload, so it should be pure. Fixed. 
I've also added exhaustive matches in backend/.